qcodes/instrument_drivers/AlazarTech/ATS.py

```diff
@@ -574,83 +614,146 @@ def acquire(self, mode=None, samples_per_record=None,
+                # otherwise continue to next buffer
+                if buffer_recycling:
+                    acquisition_controller.handle_buffer(buf.buffer, buffers_completed)
+                    self._call_dll('AlazarPostAsyncBuffer',
+                                   self._handle, ctypes.cast(buf.addr, ctypes.c_void_p), buf.size_bytes)
+                buffers_completed += 1
+                bytes_transferred += buf.size_bytes
+        finally:
+            # stop measurement here
+            done_capture = time.clock()
+            self._call_dll('AlazarAbortAsyncRead', self._handle)
+        time_done_abort = time.clock()
         # -----cleanup here-----
         # extract data if not yet done
         if not buffer_recycling:
-            for buf in self.buffer_list:
-                acquisition_controller.handle_buffer(buf.buffer)
-
+            for i, buf in enumerate(self.buffer_list):
+                acquisition_controller.handle_buffer(buf.buffer, i)
+        time_done_handling = time.clock()
         # free up memory
         self.clear_buffers()
 
+        time_done_free_mem = time.clock()
         # check if all parameters are up to date
-        for p in self.parameters.values():
-            p.get()
-
+        # Getting IDN is very slow so skip that
+        for name, p in self.parameters.items():
+            if name != 'IDN':
+                p.get()
+
+        # Compute the total transfer time, and display performance information.
+        end_time = time.clock()
+        transfer_time_sec = end_time - start
+        setup_time = done_setup - start
+        capture_time = done_capture - done_setup
+        abort_time = time_done_abort - done_capture
+        handling_time = time_done_handling - time_done_abort
+        free_mem_time = time_done_free_mem - time_done_handling
+        # print("Capture completed in %f sec" % transfer_time_sec)
+        buffers_per_sec = 0
+        bytes_per_sec = 0
+        records_per_sec = 0
+        if transfer_time_sec > 0:
+            buffers_per_sec = buffers_completed / transfer_time_sec
+            bytes_per_sec = bytes_transferred / transfer_time_sec
+            records_per_sec = (records_per_buffer *
+                               buffers_completed / transfer_time_sec)
+        logger.info("Captured %d buffers (%f buffers per sec)" %
+                     (buffers_completed, buffers_per_sec))
+        logger.info("Captured %d records (%f records per sec)" %
+                     (records_per_buffer * buffers_completed, records_per_sec))
+        logger.info("Transferred {:g} bytes ({:g} "
+                     "bytes per sec)".format(bytes_transferred, bytes_per_sec))
+        logger.info("Pre capture setup took {}".format(setup_time))
+        logger.info("Capture took {}".format(capture_time))
+        logger.info("abort took {}".format(abort_time))
+        logger.info("handling took {}".format(handling_time))
+        logger.info("free mem took {}".format(free_mem_time))
```

Comment:
Would it be sensible to add an option to disable all this timing debug info, we don't always want to delay our measurements by printing this stuff